### PR TITLE
Resolve: 14911 Update to Django 3.2

### PR DESCRIPTION
**Pull request by Afani97**
- Update requirements to latest compatible versions with Django 3.2

Closes #14911

### CODE REVIEW

File: requirements/base/base.in
Comment (copelco) [inline requirements/base/base.in:9]: I'm pretty sure @tobiasmcnulty is a maintainer on django-cache-machine:

image

Maybe we can ping him directly or get you a commit bit on it? 😄 

Alternatively, I wonder if it would be valuable to consider another cache package, but I don't know how easy it would be to switch.

File: requirements/base/base.in
Comment (Afani97) [reply requirements/base/base.in:2]: Ah yes, good catch.

File: requirements/base/base.in
Comment (Afani97) [reply requirements/base/base.in:9]: I have an open PR for a Django upgrade for that package which needs a review and if it's merged then that's great! Otherwise I wasn't able to find a comparable package yet.

**Comment by copelco** [comment]
I see a few warnings when running `runserver`:

```
❯ python manage.py runserver
Watching for file changes with StatReloader
2022-07-15 12:02:21,289 django.utils.autoreload INFO     Watching for file changes with StatReloader
Performing system checks...

System check identified some issues:

WARNINGS:
nc.Agency: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.Dataset: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.Import: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.StateFacts: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.TopAgencyFacts: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
```

**Review by copelco — COMMENTED** [review]
I see a few warnings when running `runserver`:

```
❯ python manage.py runserver
Watching for file changes with StatReloader
2022-07-15 12:02:21,289 django.utils.autoreload INFO     Watching for file changes with StatReloader
Performing system checks...

System check identified some issues:

WARNINGS:
nc.Agency: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.Dataset: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.Import: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.StateFacts: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
tsdata.TopAgencyFacts: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
        HINT: Configure the DEFAULT_AUTO_FIELD setting or the TSDataConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
```

A bunch of deprecations are listed with `python -Wa manage.py runserver`. There are some larger upgrades in our pipeline, like Celery 4.x to 5.x, but perhaps we can hold off on those until we upgrade to Django 4.x?

**Comment by Afani97** [comment]
@copelco Ok I fixed the current warnings for this version of Django, I'll hold off fixing the other dependency warnings when we get to upgrading to v4.

**Review by copelco — APPROVED** [review]

**Pull request merged**
